cartoframes/context.py

```diff
@@ -460,15 +460,15 @@ def remove_tempfile(filepath):
             """removes temporary file"""
             os.remove(filepath)
 
-        tempfile = '{temp_dir}/{table_name}.csv'.format(temp_dir=temp_dir,
-                                                        table_name=table_name)
+        tempfile = os.path.join('temp_dir, table_name + '.csv.gz')
         self._debug_print(tempfile=tempfile)
         df.drop(labels=[geom_col], axis=1, errors='ignore').to_csv(
                 path_or_buf=tempfile,
                 na_rep='',
                 header=pgcolnames,
                 index=False,
-                encoding='utf-8')
+                encoding='utf-8',
+                compression='gzip')
```

Comment:
I love this idea, but we've found mixed results in terms of performance. Do you have any benchmarks that compares the non-compressed to the compressed?

I think what we'll do here is add an optional `compression` kwarg, and then do something like follows:

```python
tempfile = os.path.join(temp_dir, '{table_name}.{ext}'.format(table_name=table_name, ext='csv.gz' if kwargs.get('compression') else 'csv')
...
df.drop(...

        compression=kwargs.get('compression')) 
```

Reply:
I don't have a micro benchmark, just a sense from using the ui. I tend to be uploading large files so perhaps other use cases will have different results. An optional arge would work.